mocks/DDsmu_mocks/countpairs_s_mu_mocks_kernels.c.src

```diff
@@ -276,16 +285,26 @@ static inline int countpairs_s_mu_mocks_avx512_intrinsics_DOUBLE(const int64_t N
             }
 
             const AVX512_MASK m_mask = m_mask_left;
-            AVX512_FLOATS m_sbin     = AVX512_SETZERO_FLOAT();
-            for(int kbin=nsbin-1;kbin>=1;kbin--) {
-                const AVX512_MASK m_bin_mask = AVX512_MASK_COMPARE_FLOATS(m_mask_left, m_sqr_s, m_supp_sqr[kbin-1],_CMP_GE_OQ);
-                m_sbin = AVX512_BLEND_FLOATS_WITH_MASK(m_bin_mask, m_sbin, AVX512_SET_FLOAT(kbin));
-                m_mask_left = AVX512_MASK_BITWISE_AND_NOT(m_bin_mask, m_mask_left);//ANDNOT(X, Y) -> NOT X AND Y
-                if(m_mask_left == 0) {
-                    break;
+            AVX512_FLOATS m_sbin = AVX512_SETZERO_FLOAT();
+
+            if (bin_type == 1) {
+                if (!need_savg) {
```

Comment:
I wanted to compute SQRT(s2) if *not computed* before (i.e. need_savg is False).
Maybe I am missing something?

Reply:
Right; the custom binning version uses `s^2`, but the linear binning uses `s`, so we need to take the sqrt, unless it was already done because `need_savg`.

Perhaps line 272 above should just be `if(need_savg || bin_type == BIN_LIN)`?

Reply:
Yup - what @lgarrison said